Cannot have paths /user/me and /user/:id/posts at the same time

Currently, it's not possible to create the following endpoints at the same time:
```
method=GET path=/user/:id
method=GET path=/user/:id/posts
```

Paths with parameters as the first part are also conflicting with any other non-parametrical paths:
```
method=GET path=/users
method=GET path=/:username/posts
```

It'd be very nice to support these kinds of paths without conflicts, would it be possible?

Sorry, no its actually not working:
```
❌ Building Encore application graph... Failed: user/user.go:130:1: invalid API path: cannot combine path segment 'me' with path '/user/:username/posts' (other declaration at /post/post.go:118:1) (and 2 more errors)
```
The repo is https://github.com/ValeriaVG/social-network,  the error is reproducible if I replace:
```go
//encore:api auth method=GET path=/user/posts/:username
func ListUserPosts
```
with:
```go
//encore:api auth method=GET path=/user/:username/posts
func ListUserPosts
```

It's conflicting with:
```go
//encore:api auth method=GET path=/user/me
func Me(ctx context.Context) (*AuthorizedUserResponse, error) {
```

--- a/docs/develop/services-and-apis.md
+++ b/docs/develop/services-and-apis.md
@@ -106,6 +106,21 @@ func UpdateBlogPost(ctx context.Context, id int, post *BlogPost) error {
 }
 ```
 
+<Callout type="important">
+You will not be able to define paths that conflict with each other, including paths
+where the static part can be mistaken for a parameter, e.g both `/blog` and `/blog/:id` would conflict with `/:username`.
+</Callout>
+
+As a rule of thumb, try to place path parameters at the end of the path and
+prefix them with the service name, e.g:
+
+```
+GET /blog/posts
+GET /blog/posts/:id
+GET /user/profile/:username
+GET /user/me
+```
+
 #### Query parameters
 When fetching data with `GET` endpoints, it's common to receive additional parameters for optional behavior, like filtering a list or changing the sort order.
 

